packages/ra-ui-materialui/src/layout/Error.js

```diff
@@ -0,0 +1,107 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import compose from 'recompose/compose';
+import Button from '@material-ui/core/Button';
+import ExpansionPanel from '@material-ui/core/ExpansionPanel';
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
+import Hidden from '@material-ui/core/Hidden';
+import { withStyles } from '@material-ui/core/styles';
+import ErrorIcon from '@material-ui/icons/Report';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import History from '@material-ui/icons/History';
+
+import AppBarMobile from './AppBarMobile';
+import { translate } from 'ra-core';
+
+const styles = theme => ({
+    container: {
+        display: 'flex',
+        flexDirection: 'column',
+        alignItems: 'center',
+        justifyContent: 'center',
+        [theme.breakpoints.down('sm')]: {
+            padding: '1em',
+        },
+        fontFamily: 'Roboto, sans-serif',
+        opacity: 0.5,
+    },
+    title: {
+        display: 'flex',
+        alignItems: 'center',
+    },
+    icon: {
+        width: '2em',
+        height: '2em',
+        marginRight: '0.5em',
+    },
+    panel: {
+        marginTop: '1em',
+    },
+    panelDetails: {
+        whiteSpace: 'pre-wrap',
+    },
+    toolbar: {
+        marginTop: '2em',
+    },
+});
+
+function goBack() {
+    history.go(-1);
+}
+
+const Error = ({
+    error,
+    errorInfo,
+    classes,
+    className,
+    translate,
+    ...rest
+}) => (
+    <Fragment>
+        <Hidden mdUp>
+            <AppBarMobile />
+        </Hidden>
+        <div className={classnames(classes.container, className)} {...rest}>
+            <h1 className={classes.title}>
```

Comment:
Accessibility: add the `role` attribute with the `alert` value.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alert_role